File: 03. Recursion/Fibonacciseries.py
""" 
fibonacci series is a series of number in which each number is the sum of its previous two numbers

ex:  0, 1, 1, 2, 3, 5, 8....................


"""

def fib(n):
  if(n == 1):
    return 0
  elif(n == 2):
    return 1
  else:
    return fib(n-2)+fib(n-1)

n = int(input("Enter the number of terms you wants to print")
for i in range(n):
        print(fib(i), end = " ")
  

File: 01. Mathematics/Fibonacci Series.cpp
/* 
A series of numbers in which each number ( Fibonacci number ) is the sum of the two preceding numbers. 
The simplest is the series 1, 1, 2, 3, 5, 8, etc.
Here is simplest C++ program for Fibonacci series
*/
#include <iostream>
using namespace std;
int main(){
	int first=0;
	int second=1;
	int temp,num;
	cout<<"Enter a number: ";
	cin>>num;
	cout<<"Here is Fibonacci Series till "<<num<<endl;
	cout<<first<<endl<<second<<endl;
	for(int i=0; i<num-2; i++)
	{
		
	temp=first +second;
	    first=second;
	    second=temp;
		
		cout<<temp<<endl;
		
	}
}
